cache/remotecache/export.go

```diff
@@ -75,20 +76,23 @@ func (ce *contentCacheExporter) Finalize(ctx context.Context) (map[string]string
 	}
 
 	// own type because oci type can't be pushed and docker type doesn't have annotations
-	type manifestList struct {
+	type abstractManifest struct {
 		specs.Versioned
 
-		MediaType string `json:"mediaType,omitempty"`
-
+		MediaType string               `json:"mediaType,omitempty"`
+		Config    *ocispecs.Descriptor `json:"config,omitempty"`
 		// Manifests references platform specific manifests.
-		Manifests []ocispecs.Descriptor `json:"manifests"`
+		Manifests []ocispecs.Descriptor `json:"manifests,omitempty"`
+		Layers    []ocispecs.Descriptor `json:"layers,omitempty"`
 	}
 
-	var mfst manifestList
+	var mfst abstractManifest
 	mfst.SchemaVersion = 2
 	mfst.MediaType = images.MediaTypeDockerSchema2ManifestList
-	if ce.oci {
+	if ce.oci && !ce.imageManifest {
```

Comment:
We should still respect the value of `oci-mediatypes` here, even when `image-manifest=true`.

The default for `oci-mediatypes` is false, so it will still work out of the box with ECR, but users may use other registries and want to use OCI media types instead, so we should give them the option to enable this.

Reply:
Probably `oci-mediatypes=false` and `image-manifest=true` would be invalid configuration and should just error.

What would be a better name for `image-manifest=true`? Is it `oci-artifact-manifest` `oci-artifact`? I'm quite confused about what they mean, but `image-manifest` seems too generic and doesn't convey the meaning that this uses a config with custom mediatyoe.

Reply:
Funnily enough in my initial work for this I did use `oci-artifact` as the key name here, but after working with some colleagues more closely associated with the OCI image-spec 1.1 work ( tagging @michaelb990 and @jlbutler ), we found that the term `artifact` has become semantically loaded. More specifically there's two artifacts in people's mind in the space:
- small "a" artifacts: image manifest + layers with a nonstandardized `config.mediaType` (this aligns with the cache manifest approach in this PR)
- capital "A" Artifacts: artifact manifest + blobs (currently being pulled out / made optional in 1.1). This is an approach we could take for cache manifest if/when this manifest type becomes more prevalent.

As I work through this happy to consider other names for the key. 

Will think through oci-mediatypes: I think I lean on erroring here though.

Reply:
Thinking about it more: ImageManifest mode _could_ support docker's image manifest version (which I don't think makes sense given registry support is wide enough for OCI manifest), so I am going to error if oci-mediatypes=false and image-manifest=true as invalid and down the road that can be implemented as needed/warranted. I also will update the readme to instruct client users to need to toggle both for Image Manifest remote caching to work.